### PR TITLE
organization cli and api tests refactored

### PR DESCRIPTION
**Pull request by pondrejk**
merged and removed tests based on https://mojo.redhat.com/groups/satellite6qe/blog/2019/06/04/making-robottelo-slimmer

Test results:
```
pytest tests/foreman/cli/test_organization.py
======================================================= test session starts                                                          ========================================================
platform linux -- Python 3.7.3, pytest-4.0.2, py-1.7.0, pluggy-0.8.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/pondrejk/Documents/robottelo, inifile:
plugins: xdist-1.25.0, services-1.3.1, mock-1.10.4, forked-0.2, cov-2.6.1
collecting ... 2019-06-13 16:44:27 - conftest - DEBUG - BZ deselect is disabled in settings

collected 17                                                                                                                         items

tests/foreman/cli/test_organization.py .................

============================================= 17 passed, 57 warnings in 853.70 seconds                                               =============================================
```
```
pytest tests/foreman/api/test_organization.py
==================================================== test session starts ====================================================
platform linux -- Python 3.7.3, pytest-4.0.2, py-1.7.0, pluggy-0.8.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/pondrejk/Documents/robottelo, inifile:
plugins: xdist-1.25.0, services-1.3.1, mock-1.10.4, forked-0.2, cov-2.6.1
collecting ... 2019-06-19 18:07:07 - conftest - DEBUG - BZ deselect is disabled in settings

collected 14 items

tests/foreman/api/test_organization.py ..............                                                                 [100%]

================================================ 14 passed in 126.01 seconds ================================================
```

I tired to sum up the changes in respective commit names, the highest gain comes form reducing the unnecessary valid-names-list circulation for create and update tests.
For comparison, test result on before this change:

API: 2 failed, 20 passed, 6 warnings in 282.47 seconds
CLI: 2 failed, 35 passed, 6 warnings in 4043.76 seconds
The execution time for api and cli combined is now reduced to 8.35 %

### CODE REVIEW

File: tests/foreman/cli/test_organization.py [hunk 1]
```diff
@@ -115,142 +113,62 @@ def test_verify_bugzilla_1078866(self):
...
-                    'description': desc,
-                    'name': name,
-                })
-                self.assertEqual(org['name'], name)
-                self.assertEqual(org['description'], desc)
-
-    @tier1
-    @upgrade
-    def test_positive_create_with_name_label_description(self):
+    def test_positive_CRD(self):
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_organization.py]: typo `CRD` => `CRUD` ?

File: tests/foreman/cli/test_organization.py [hunk 2]
```diff
@@ -115,142 +113,62 @@ def test_verify_bugzilla_1078866(self):
...
+        name = valid_org_names_list()[0]
+        label = valid_labels_list()[0]
+        desc = valid_data_list()[0]
+        org = make_org({
+            'name': name,
+            'label': label,
+            'description': desc,
+            })
+        self.assertEqual(org['name'], name)
+        self.assertNotEqual(org['name'], org['label'])
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_organization.py]: this assertion is required? my point is here that valid_org_names_list()[0] can be equal to valid_labels_list()[0] in any case ?

File: tests/foreman/cli/test_organization.py [hunk 3]
```diff
@@ -115,142 +113,62 @@ def test_verify_bugzilla_1078866(self):
...
+        self.assertEqual(org['label'], label)
+        self.assertEqual(org['description'], desc)
 
-        :BZ: 1259374
+        # List
+        result = Org.list({'search': 'name=%s' % org['name']})
+        self.assertTrue(len(result) > 0)
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_organization.py]: will below code add more value, what you think ?
 
` self.assertTrue(len(result) == 2)` assuming default org will exist.

File: tests/foreman/cli/test_organization.py (hunk 1 above)
Comment (pondrejk) [reply tests/foreman/cli/test_organization.py]: well, I do the update tests separately (including also those would make the test too long), therefore no U :)

**Comment by pondrejk** [comment]
@omkarkhatavkar  good points, updated, thanks

File: tests/foreman/api/test_organization.py [hunk 4]
```diff
@@ -478,69 +341,3 @@ def test_negative_update(self):
                         id=self.organization.id,
                         **attrs
                     ).update(attrs.keys())
-
-    @tier2
-    @upgrade
-    @skip_if_bug_open('bugzilla', 1103157)
-    def test_verify_bugzilla_1103157(self):
```
Comment (ntkathole) [inline tests/foreman/api/test_organization.py]: @pondrejk This test looks good. Why we re removing it? Is it already covered somewhere?

Comment (pondrejk) [reply tests/foreman/api/test_organization.py]: yeah I was a bit hesitant about this one, adding multiple compute resources and checking there is no conflict is test_positive_add_and_remove_compresources in cli. This test does it non-admin user, but I didn't feel it justifies a separate test case. Anyway, if there's a strong opinion to keep it in, I'm ok with that.

Comment (ntkathole) [reply tests/foreman/api/test_organization.py]: I am ok with it if same scenario without non-admin user is covered in another test. Thanks.

**Pull request merged**